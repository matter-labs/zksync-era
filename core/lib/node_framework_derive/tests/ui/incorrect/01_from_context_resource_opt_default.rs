use zksync_node_framework_derive::FromContext;

mod service {
    pub struct ServiceContext<'a> {
        _marker: std::marker::PhantomData<&'a ()>,
    }

    impl<'a> ServiceContext<'a> {
        pub fn get_resource<T>(&mut self) -> Result<T, crate::WiringError> {
            unimplemented!()
        }

        pub fn get_resource_or_default<T>(&mut self) -> Result<T, crate::WiringError> {
            unimplemented!()
        }
    }
}

pub enum WiringError {
    ResourceLacking { _name: String },
}

pub trait FromContext: Sized {
    fn from_context(context: &mut service::ServiceContext<'_>) -> Result<Self, WiringError>;
}

#[derive(FromContext)]
#[ctx(local)]
struct StructWithOption {
    #[resource(default)]
    _field: Option<u8>,
}

fn main() {}
