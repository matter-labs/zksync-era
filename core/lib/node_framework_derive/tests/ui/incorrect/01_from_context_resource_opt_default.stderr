error: `default` cannot be used on `Option` types
  --> tests/ui/incorrect/01_from_context_resource_opt_default.rs:31:5
   |
31 |     _field: Option<u8>,
   |     ^^^^^^
